test/AngularSpec.js

```diff
@@ -2058,6 +2058,10 @@ describe('angular', function() {
     it('should serialize undefined as undefined', function() {
       expect(toJson(undefined)).toEqual(undefined);
     });
+
+    it('should serialize invalid dates to null', function() {
+      expect(toJson(new Date(1 / 0))).toEqual('null');
```

Comment:
To cover more cases, put the invalid date inside an object (e.g. `toJson({when: new Date(1 / 0)})`).